relay/purger/purger.go

```diff
@@ -54,6 +54,7 @@ const (
 
 // Purger purges relay log according to some strategies
 type Purger struct {
+	sync.RWMutex
```

Comment:
may use a named field is better (like `lock sync.RWMutex`), then outer code can't call methods from `RWMutex`. other similar usages can be updated later.